types/oidcsettings.pp

```diff
@@ -1,113 +1,148 @@
+    Optional['CacheDir']                                   => String,
+    Optional['CacheEncrypt']                               => Enum['On', 'Off'],
+    Optional['CacheShmMax']                                => Integer,
+    Optional['CacheShmEntrySizeMax']                       => Integer,
+    Optional['CacheFileCleanInterval']                     => Integer,
+    Optional['MemCacheServers']                            => String,
+    Optional['MemCacheConnectionsHMax']                    => Integer,
+    Optional['MemCacheConnectionsMin']                     => Integer,
+    Optional['MemCacheConnectionsSMax']                    => Integer,
+    Optional['MemCacheConnectionsTTL']                     => Integer,
+    Optional['RedisCacheServer']                           => String,
+    Optional['RedisCachePassword']                         => String,
+    Optional['RedisCacheConnectTimeout']                   => Pattern[/^[0-9]+\s?[0-9]*$/],
+    Optional['RedisCacheDatabase']                         => Integer,
+    Optional['RedisCacheTimeout']                          => Integer,
+    Optional['RedisCacheUsername']                         => String,
+    Optional['DiscoverURL']                                => Variant[Stdlib::HTTPSUrl, Stdlib::HttpUrl],
+    Optional['HTMLErrorTemplate']                          => String,
+    Optional['DefaultURL']                                 => Variant[Stdlib::HTTPSUrl, Stdlib::HttpUrl],
+    Optional['PathScope']                                  => Pattern[/^\"?[A-Za-z0-9\-\._\s]+\"?$/],
+    Optional['PathAuthRequestParams']                      => Pattern[/^[A-Za-z0-9\-\._%]+=[A-Za-z0-9\-\._%]+(&[A-Za-z0-9\-\._%]+=[A-Za-z0-9\-\._%]+)*$/],
+    Optional['IDTokenIatSlack']                            => Integer,
+    Optional['ClaimPrefix']                                => String,
+    Optional['ClaimDelimiter']                             => Pattern[/^.$/],
+    Optional['RemoteUserClaim']                            => String,
+    Optional['PassIDTokenAs']                              => Pattern[/^((claims|payload|serialized)\s?)+$/],
+    Optional['PassUserInfoAs']                             => Pattern[/^((claims|json(:([A-Za-z0-9\-\._])+)?|(signed_)?jwt(:([A-Za-z0-9\-\._])+)?)\s?)+$/],
+    Optional['PassClaimsAs']                               => Enum['none', 'headers', 'environment', 'both'],
+    Optional['AuthNHeader']                                => String,
+    Optional['HTTPTimeoutLong']                            => Integer,
+    Optional['HTTPTimeoutShort']                           => Integer,
+    Optional['StateTimeout']                               => Integer,
+    Optional['ScrubRequestHeaders']                        => Enum['On', 'Off'],
+    Optional['OutgoingProxy']                              => String,
+    Optional['UnAuthAction']                               => Pattern[/^(auth|pass|401|407|410)\s.*/],
```

Comment:
Make the space after the first match optional so that `"401`" is accepted:

```suggestion
    Optional['UnAuthAction']                               => Pattern[/^(auth|pass|401|407|410)(\s.*)?$/],
```